src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift

```diff
@@ -0,0 +1,229 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import Foundation
+import UIKit
+
+protocol ExposureSubmissionCoordinator {
+
+	// MARK: - Attributes.
+
+	/// Delegate that is called for life-cycle events of the coordinator.
+	var delegate: ExposureSubmissionCoordinatorDelegate? { get set }
+
+	// MARK: - Navigation.
+
+	/// Starts the coordinator and displays the initial root view controller.
+	func start(with result: TestResult?)
+	func dismiss()
+
+	func showOverviewScreen()
+	func showTestResultScreen(with result: TestResult)
+	func showHotlineScreen()
+	func showTanScreen()
+	func showQRScreen(qrScannerDelegate: ExposureSubmissionQRScannerDelegate)
+	func showWarnOthersScreen()
+	func showThankYouScreen()
+}
+
+/// This delegate allows a class to be notified for life-cycle events of the coordinator.
+protocol ExposureSubmissionCoordinatorDelegate: class {
+	func exposureSubmissionCoordinatorWillDisappear(_ coordinator: ExposureSubmissionCoordinator)
+}
+
+/// Marker protocol that ensures that a `coordinator` can be set for a view controller.
+protocol ExposureSubmissionCoordinatorViewController {
+	var coordinator: ExposureSubmissionCoordinator? { get set }
+}
+
+/// Concrete implementation of the ExposureSubmissionCoordinator protocol.
+class ESCoordinator: ExposureSubmissionCoordinator {
+
+	// MARK: - Attributes.
+
+	weak var delegate: ExposureSubmissionCoordinatorDelegate?
+	weak var parentNavigationController: UINavigationController?
+	weak var navigationController: UINavigationController?
+
+	/// - NOTE: We need a strong (aka non-weak) reference here.
+	var exposureSubmissionService: ExposureSubmissionService?
```

Comment:
why do you set `ExposureSubmissionService` as an optional type? And also set it to `var` instead of `let`? Will you change it later

Reply:
thanks I would have forgotten about that! :)